packages/binding-netconf/src/codecs/netconf-codec.ts

```diff
@@ -70,32 +75,30 @@ export default class NetconfCodec {
             } else {
                 parsed = value;
             }
+
+            return parsed;
             // TODO check the schema!
         } catch (err) {
             if (err instanceof SyntaxError) {
                 if (bytes.byteLength === 0) {
                     // empty payload -> void/undefined
-                    parsed = undefined;
+                    return undefined;
                 } else {
                     // be relaxed about what is received -> string without quotes
-                    parsed = bytes.toString();
+                    return bytes.toString();
                 }
             } else {
                 throw err;
             }
         }
-
-        return parsed;
     }
 
-    valueToBytes(value: any, schema: TD.DataSchema, parameters?: { [key: string]: string }): Buffer {
+    valueToBytes(value: unknown, schema: TD.DataSchema, parameters?: { [key: string]: string }): Buffer {
         // console.debug("NetconfCodec serializing", value);
         let body = "";
         if (value !== undefined) {
             const NSs = {};
-            // let leaf = value.leaf;
-            let leaf = <any>schema;
-            leaf = leaf.forms[0].href.split("/").splice(-1, 1); // take the first one, since there is no difference for the leaf
+            let leaf = schema.forms[0].href.split("/").splice(-1, 1); // take the first one, since there is no difference for the leaf
```

Comment:
This part of the method seems to be broken at the moment as it only works with property affordances and not with generic dataschemas.